packages/lib/services/rest/routes/resources.ts

```
@@ -62,7 +64,11 @@ export default async function(request: Request, id: string = null, link: string
 		}
 
 		if (isUpdate && !id) throw new ErrorBadRequest('Missing resource ID');
-		const filePath = request.files[0].path;
+		// Prevent path traversal: extract only the filename portion and
+		// validate it against a strict whitelist before use.
+		const fileName = path.basename(request.files[0].path);
+		if (!fileName || !/^[\w.\-]+$/.test(fileName)) throw new ErrorBadRequest('Invalid file path');
```

ESLint

[error] Unnecessary escape character: \-.

(no-useless-escape)